crates/node/src/interactive_cli/context.rs

```diff
@@ -163,22 +176,64 @@ impl ContextCommand {
             Commands::Join {
                 private_key,
                 invitation_payload,
+                name,
+                author,
             } => {
+                let public_key = private_key.public_key();
+
+                // Validate --as alias if provided
+                if let Some(author_alias) = &author {
+                    let resolved_pk = node
+                        .ctx_manager
+                        .resolve_alias(author_alias.clone(), None)?
+                        .ok_or_eyre(format!("alias '{}' not found", author_alias))?;
+                    if resolved_pk != public_key {
+                        return Err(eyre::eyre!(
+                            "alias '{}' does not match the private key's public key {}",
+                            author_alias,
+                            public_key
+                        ));
+                    }
+                }
                 let response = node
                     .ctx_manager
                     .join_context(private_key, invitation_payload)
                     .await?;
-
-                if let Some((context_id, identity)) = response {
-                    println!(
-                        "{ind} Joined context {context_id} as {identity}, waiting for catchup to complete..."
-                    );
-                } else {
-                    println!(
-                        "{ind} Unable to join context at this time, a catchup is in progress."
-                    );
-                }
+                    if let Some((context_id, identity)) = response {
+                        // Create identity alias in the context if --name is provided
+                        if let Some(name_alias) = name {
+                            node.ctx_manager
+                                .create_alias(name_alias.as_ref(), Some(context_id), identity)?;
+                            println!(
+                                "{} Created identity alias '{}' for {} in context {}",
```

Comment:
```suggestion
                                "{} Created identity alias '{}' for '{}' in context '{}'",
```

and other places you have `print`

Reply:
this isn't yet resolved